packages/opentelemetry-instrumentation-fetch/src/fetch.ts

```diff
@@ -158,6 +158,11 @@ export class FetchInstrumentation extends InstrumentationBase<
       api.propagation.inject(api.context.active(), options.headers, {
         set: (h, k, v) => h.set(k, typeof v === 'string' ? v : String(v)),
       });
+    } else if(options.headers instanceof Headers) {
+      // ||-operator causes ts type error
```

Comment:
What is this comment referring to?

Reply:
I tried to use the `||`-operator inside the `if`-statement to avoid the `else if` and the code duplication. But it caused a type error at the `setter`-parameter of the `api.propagation.inject`-function.